open-metadata-implementation/adapters/open-connectors/repository-services-connectors/open-metadata-collection-store-connectors/inmemory-repository-connector/src/main/java/org/odpi/openmetadata/adapters/repositoryservices/inmemory/repositoryconnector/InMemoryEntityNeighbourhood.java

```diff
@@ -0,0 +1,349 @@
+     *                             status values.
+     * @param limitResultsByClassification List of classifications that must be present on all returned entities.
+     * @param level the number of the relationships out from the starting entity that the query will traverse to
+     *              gather results.
+     * @throws RepositoryErrorException there is a problem communicating with the metadata repository where
+     *                                  the metadata collection is stored.
+     * @throws InvalidParameterException one of the parameters is invalid or null.
+     */
+    public InMemoryEntityNeighbourhood(
+                                       OMRSRepositoryValidator repositoryValidator,
+                                       Map<String, EntityDetail>   entityStore,
+                                       Map<String, Relationship>   relationshipStore,
+                                       String               rootEntityGUID,
+                                       List<String>         entityTypeGUIDs,
+                                       List<String>         relationshipTypeGUIDs,
+                                       List<InstanceStatus> limitResultsByStatus,
+                                       List<String>         limitResultsByClassification,
+                                       int                  level) throws RepositoryErrorException, InvalidParameterException
+    {
+        this.repositoryValidator =repositoryValidator;
+        this.entityStore = entityStore;
+        this.relationshipStore = relationshipStore;
+        this.rootEntityGUID = rootEntityGUID;
+        this.entityTypeGUIDs =entityTypeGUIDs;
+        this.relationshipTypeGUIDs = relationshipTypeGUIDs;
+        this.limitResultsByStatus =limitResultsByStatus;
+        this.limitResultsByClassification=limitResultsByClassification;
+        // limit the level
+        if (level <1 || level >10 ) {
+            level =10;
+        }
+
+        this.level = level;
+
+        initializeMaps();
+
+    }
+
+    /**
+     * Initialize maps that help us traverse between entities and relationships using their guids
+     *
+     * @throws RepositoryErrorException there is a problem communicating with the metadata repository where
+     *                                  the metadata collection is stored.
+     * @throws InvalidParameterException one of the parameters is invalid or null.
+     */
+    private void initializeMaps() throws RepositoryErrorException, InvalidParameterException
+    {
+        for (Relationship relationship : relationshipStore.values())
+        {
+            String relationshipGuid = relationship.getGUID();
+            String relationshipEnd1Guid = getEnd1EntityGUID(relationship);
+            String relationshipEnd2Guid = getEnd2EntityGUID(relationship);
+//            EntityDetail entity1 = entityStore.get(relationshipEnd1Guid);
```

Comment:
Why is this commented out - its a new file

Reply:
fixed

Reply:
removed